test/box-luatest/downgrade_test.lua

```diff
@@ -0,0 +1,799 @@
+            "It is supported starting from version 2.11.0.",
+        })
+    end)
+end
+
+-- Manual check
+--
+--  Tarantool Enterprise 2.11.0-entrypoint-113-g803baaf
+--  type 'help' for interactive help
+--  tarantool> box.cfg{auth_type = 'pap-sha256'}
+--  tarantool> box.schema.user.create('alice', {password = 'secret'})
+--  tarantool> box.schema.downgrade('2.10.0')
+--  ---
+--  - error: 'builtin/box/upgrade.lua:1861: Auth type ''pap-sha256'' is found
+--      for user ''alice''. It is supported starting from version 2.11.0.'
+--  ...
+g.test_downgrade_non_chap_sha1_auth = function(cg)
+    cg.server:exec(function()
+        local helper = require('test.box-luatest.downgrade_helper')
+
+        -- ALICE password with some salt
+        local auth = {
+            ['pap-sha256'] = {
+                'WOi6OgOXNUwxr0YU1sR9/rzJYnU=',
+                'ZfRPND3Q0ls0hmKuzcPKBJngvDrJx6cooa8HEjj4MH0=',
+            },
+        }
+        box.space._user:auto_increment{1, 'alice', 'user', auth, {}, os.time()}
+        -- BOB password with some salt
+        auth = {
+            ['pap-sha256'] = {
+                'FIXvGZk8bQmNztFmsCeTlpSxobk=',
+                '87wjwffInpRGMBNATigbXcmKMC9sdwgplKZRjqBcdxQ=',
+            },
+        }
+        box.space._user:auto_increment{1, 'bob', 'user', auth, {}, os.time()}
+
+        helper.check_issues('2.11.0', {
+            "Auth type 'pap-sha256' is found for user 'alice'. " ..
+            "It is supported starting from version 2.11.0.",
+            "Auth type 'pap-sha256' is found for user 'bob'. " ..
+            "It is supported starting from version 2.11.0.",
+        })
+    end)
+end
+
+-- No manual check required
+g.test_downgrade_user_auth_history_and_last_modified = function(cg)
+    cg.server:exec(function()
+        local helper = require('test.box-luatest.downgrade_helper')
+
+        local spaces_ids = {box.space._user.id, box.space._vuser.id}
+        local function check_has_new_auth_fields()
+            for _, row in box.space._user:pairs() do
+                t.assert_equals(#row, 7)
+                t.assert_equals(row.auth_history, {})
+                t.assert_equals(row.last_modified, 0)
+            end
+            for _, id in pairs(spaces_ids) do
+                local row = box.space._space:get(id)
+                t.assert_equals(row[7][6],
+                                {name = 'auth_history', type = 'array'})
+                t.assert_equals(row[7][7],
+                                {name = 'last_modified', type = 'unsigned'})
+            end
+        end
+
+        local function check_NO_new_auth_fields()
+            for _, row in box.space._user:pairs() do
+                t.assert_equals(#row, 5)
+            end
+            for _, id in pairs(spaces_ids) do
+                local row = box.space._space:get(id)
+                t.assert_equals(row[7][6], nil)
+                t.assert_equals(row[7][7], nil)
+            end
+        end
```

Comment:
Nit: you could make one function that returns a boolean instead of two. And magic numbers again: I'm okay with `[7][{6, 7}]` here, but what are `7`, `[7]` and `5`?